File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static public int tar = 0;
    static public int answer = 0;
    public int solution(int[] numbers, int target) {
        tar = target;
        bfs(0, 0, numbers);
        return answer;
    }
    public static void bfs(int sum, int index, int[] numbers){
      if(index == numbers.length){
        if(sum == tar) answer++;
      }
      else {
        bfs(sum+numbers[index], index+1, numbers);
        bfs(sum-numbers[index], index+1, numbers);
      }
    
  }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    const dfs = (num, i) => {        
        if (i === numbers.length) {
            if (target === num) answer++;
            return;
        }
        
        dfs(num + numbers[i], i + 1);
        dfs(num - numbers[i], i + 1);
    }
    
    dfs(0, 0);
    
    return answer;
}